### PR TITLE
[vectorized](function) support array_first_index function

### PR DESCRIPTION
**Pull request by gitccl**
# Proposed changes

Issue Number: close #xxx

## Problem summary

```
mysql> select array_first_index(x->x+1>3, [2, 3, 4]);
+-------------------------------------------------------------------+
| array_first_index(array_map([x] -> x(0) + 1 > 3, ARRAY(2, 3, 4))) |
+-------------------------------------------------------------------+
|                                                                 2 |
+-------------------------------------------------------------------+

mysql> select array_first_index(x -> x is null, [null, 1, 2]);
+----------------------------------------------------------------------+
| array_first_index(array_map([x] -> x(0) IS NULL, ARRAY(NULL, 1, 2))) |
+----------------------------------------------------------------------+
|                                                                    1 |
+----------------------------------------------------------------------+

mysql> select array_first_index(x->power(x,2)>10, [1, 2, 3, 4]);
+---------------------------------------------------------------------------------+
| array_first_index(array_map([x] -> power(x(0), 2.0) > 10.0, ARRAY(1, 2, 3, 4))) |
+---------------------------------------------------------------------------------+
|                                                                               4 |
+---------------------------------------------------------------------------------+
```

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [X] Has document been added or modified
* [ ] Does it need to update dependencies
* [X] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by gitccl** [comment]
run buildall

File: gensrc/script/doris_builtins_functions.py [hunk 1]
```diff
@@ -650,6 +649,8 @@
     [['array_exists'], 'ARRAY_BOOLEAN', ['ARRAY_VARCHAR'], ''],
     [['array_exists'], 'ARRAY_BOOLEAN', ['ARRAY_STRING'], ''],
 
+    [['array_first_index'], 'BIGINT', ['ARRAY_BOOLEAN'], 'ALWAYS_NULLABLE'],
+
```
Comment (zhangstar333) [inline gensrc/script/doris_builtins_functions.py:653]: this function seems better to return 0 if it is not found in any case, so maybe set always_not_nullable mode

File: fe/fe-core/src/main/java/org/apache/doris/analysis/LambdaFunctionCallExpr.java [hunk 2]
```diff
@@ -133,7 +135,6 @@ public void analyzeImpl(Analyzer analyzer) throws AnalysisException {
                 LOG.warn("fn {} not exists", this.toSqlImpl());
                 throw new AnalysisException(getFunctionNotFoundError(collectChildReturnTypes()));
             }
-            fn.setReturnType(getChild(0).getType());
         } else if (fnName.getFunction().equalsIgnoreCase("array_filter")) {
```
Comment (zhangstar333) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/LambdaFunctionCallExpr.java:138]: why remove this line?

Comment (gitccl) [reply fe/fe-core/src/main/java/org/apache/doris/analysis/LambdaFunctionCallExpr.java:138]: The return type has already been specified in the `fn` when we create function in `doris_builtins_functions.py`. In this case, the return type of `fn` is `Int64`, but the the return type of children 0 is `Array(Bool)`, we shouldn't change the return type of the `fn`.

**Comment by gitccl** [comment]
run buildall

File: gensrc/script/doris_builtins_functions.py (hunk 1 above)
Comment (gitccl) [reply gensrc/script/doris_builtins_functions.py:653]: done

**Review by zhangstar333 — DISMISSED** [review]

**Comment by gitccl** [comment]
run buildall

**Comment by gitccl** [comment]
run p0

**Comment by gitccl** [comment]
run buildall

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**